src/core/fonts.js

```diff
@@ -1409,7 +1408,7 @@ var Font = (function FontClosure() {
         tables["post"] = null;
 
         for (let i = 0; i < numTables; i++) {
-          const table = readTableEntry(font);
+          const table = readTableEntry(file);
```

Comment:
This one actually looks like it caused a bug since we always read the entry for the `font` variable of the upper scope here and not for the given `file` parameter. If that's indeed the case, I'm surprised we haven't seen any problems because of it, but it would show that deshadowing is a good thing!

Reply:
Yes, this case is absolutely confusing!

The only reason that this hasn't caused any bugs is that these properties are Objects (in this case `instanceof Stream`), and those are essentially passed by reference rather than value in JavaScript (the names `font`/`file` thus point to the same data).